src/main.ts

```diff
@@ -261,21 +261,26 @@ export class ConnectionFeature extends SqlOpsFeature<undefined> {
 		};
 
 		let registerOnConnectionComplete = (handler: (connSummary: azdata.ConnectionInfoSummary) => any): void => {
-			client.onNotification(protocol.ConnectionCompleteNotification.type, handler);
+			client.registerNotificationListener(protocol.ConnectionCompleteNotification.type.method, handler);
```

Comment:
Can you just not store the handlers in the feature directly, rather than modifying the client type

Reply:
You mean an array of handlers in each `let someMethodWithOnNotification` ?

That's what I was initially planning to do, but it was a lot of boilerplate code per method.